### PR TITLE
[plugin.video.akibapass] v0.5.0

### PR DESCRIPTION
**Pull request by MrKrabat**
### Description
Akibapass.de is a paid anime streaming service offering titles from four german publisher.
Akibapass is available in germany, austria and switzerland with german sub and dub.
A paid subscription costs 8€/month, otherwise it is possible to buy anime for 2€/episode which allows downloading.

~~Note: the server.py is needed for this reason:
The stream manifest is a one-use link but kodi access its 4 times.
If we download it as a local file, we could not pass any cookies.
That's why we need a local server to pass it to kodi.~~

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.akibapass/resources/lib/view.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+def add_item(args, info, isFolder=True, total_items=0, queued=False, rex=re.compile(r"(?<=mode=)[^&]*"), mediatype="video"):
+    """Add item to directory listing.
+    """
+    # Defaults in dict. Use "None" instead of None so it is compatible for
+    info = set_info_defaults(args, info)
+    u = build_url(info)
+
+    # Create list item
+    li = xbmcgui.ListItem(label          = info["title"],
```
Comment (Lunatixz) [inline plugin.video.akibapass/resources/lib/view.py]: Can we move this to setart? IconImage and thumbnailImage were deprecated jarvis and up :)
http://mirrors.xbmc.org/docs/python-docs/16.x-jarvis/xbmcgui.html#ListItem

File: plugin.video.akibapass/resources/lib/server.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        sock.setblocking(0)
+    except socket.error:
+        # unable to listen on port
+        xbmc.log("[SERVICE] Akibapass: Failed listening on port 10147", xbmc.LOGFATAL)
+        return
+
+    timer   = time.time() + 10
+    counter = 0
+    while (time.time() < timer) and (counter < 4):
+        time.sleep(0.03)
```
Comment (Lunatixz) [inline plugin.video.akibapass/resources/lib/server.py]: please change to xbmc.sleep()
http://mirrors.xbmc.org/docs/python-docs/16.x-jarvis/xbmc.html#-sleep

File: plugin.video.akibapass/resources/lib/netapi.py [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+            # play stream
+            item = xbmcgui.ListItem(args.name, path="http://localhost:10147/stream.m3u8" + login.getCookie(args))
+            item.setInfo(type="Video", infoLabels={"Title":       args.name,
+                                                   "TVShowTitle": args.name,
+                                                   "episode":     args.episode,
+                                                   "rating":      args.rating,
+                                                   "plot":        args.plot,
+                                                   "year":        args.year,
+                                                   "studio":      args.studio})
+            item.setThumbnailImage(args.icon)
```
Comment (Lunatixz) [inline plugin.video.akibapass/resources/lib/netapi.py]: Deprecated

File: plugin.video.akibapass/resources/lib/netapi.py [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+            m3u8 = urllib2.urlopen(url)
+            m3u8 = m3u8.read()
+
+            # start stream provider
+            t = threading.Thread(target=server.streamprovider, args=(str(m3u8),))
+            t.start()
+            xbmc.sleep(50)
+
+            # play stream
+            item = xbmcgui.ListItem(args.name, path="http://localhost:10147/stream.m3u8" + login.getCookie(args))
```
Comment (Lunatixz) [inline plugin.video.akibapass/resources/lib/netapi.py]: Can someone double check these local redirects...

**Comment by MrKrabat** [comment]
Thanks for reviewing. I've updated the commit.

**Comment by MrKrabat** [comment]
I am super happy right now! :)
I found a way that kodi will open the URL only one time so these dirty redirects are not needed any more!
```
item.setMimeType("application/vnd.apple.mpegurl")
item.setContentLookup(False)
```
I will prepare a update for this

**Pull request merged**